test/common/grpc/grpc_client_integration_test.cc

```diff
@@ -293,6 +293,27 @@ TEST_P(GrpcClientIntegrationTest, ResourceExhaustedError) {
   dispatcher_helper_.runDispatcher();
 }
 
+// Validate that a trailers Unauthenticated reply is handled.
+TEST_P(GrpcClientIntegrationTest, UnauthenticatedError) {
+  initialize();
+  auto stream = createStream(empty_metadata_);
+  stream->sendServerInitialMetadata(empty_metadata_);
+  stream->sendServerTrailers(Status::GrpcStatus::Unauthenticated, "error message", empty_metadata_);
+  dispatcher_helper_.runDispatcher();
+}
+
+// Validate that a trailers reply is still handled even if a grpc status code larger than
+// MaximumValid, is handled.
+TEST_P(GrpcClientIntegrationTest, MaximumValidPlusOne) {
+  SKIP_IF_GRPC_CLIENT(ClientType::EnvoyGrpc);
```

Comment:
Why skip this test for the Envoy client? Shouldn't it also work fine?

Reply:
Sorry I would have to look into this in more detail. At a functional level, it seems like this test should work for EnvoyGrpc if the other test works for EnvoyGrpc. If it doesn't work, I think it's likely a bug in EnvoyGrpc since we should handle unknown things there also?

Reply:
I agree. 

-1 is harmful at the client level for google grpc client, but not the callback level (onRemoteClose).
-1 might not be harmful to envoy grpc client.